hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/HoodieStreamerUtils.java

```diff
@@ -118,7 +119,7 @@ public static Option<JavaRDD<HoodieRecord>> createHoodieRecords(HoodieStreamer.C
                   if (shouldErrorTable) {
                     Schema schema = genRec.getSchema();
                     if (!GenericData.get().validate(schema, genRec)) {
-                      throw new HoodieException("The record does not match the schema: " + schema);
+                      throw new HoodieIOException("The record to be serialized does not match the schema: " + schema);
```

Comment:
don't we need to fix any of validations in test after this fix?

Reply:
New tests are added to cover the logic.